src/vma/dev/ring_simple.cpp

```diff
@@ -360,34 +360,24 @@ bool ring_simple::attach_flow(flow_tuple& flow_spec_5t, pkt_rcvr_sink *sink)
 	rfs* p_rfs;
 	rfs* p_tmp_rfs = NULL;
 	sockinfo* si = static_cast<sockinfo*> (sink);
-	uint32_t flow_tag_id = 0; // spec will not be attached to rule
+
+	if (si == NULL)
+		return false;
+
+	uint32_t flow_tag_id = si->get_flow_tag_val(); // spec will not be attached to rule
 
 	ring_logdbg("flow: %s, with sink (%p), m_flow_tag_enabled: %d",
 		    flow_spec_5t.to_str(), si, m_flow_tag_enabled);
 
-	if( si == NULL )
-		return false;
-
 	// If m_flow_tag_enabled==true then flow tag is supported and flow_tag_id is guaranteed
 	// to have a !0 value which will results in a flow id being added to the flow spec.
 	// Otherwise, flow tag is not supported, flow_tag_id=0 and no flow id will be set in the flow spec.
 	if (m_flow_tag_enabled) {
 		// sockfd=0 is valid too but flow_tag_id=0 is invalid, increment it
 		// effectively limiting our sockfd range to FLOW_TAG_MASK-1
-		int flow_tag_id_candidate = si->get_fd() + 1;
-		if (flow_tag_id_candidate > 0) {
-			flow_tag_id = flow_tag_id_candidate & FLOW_TAG_MASK;
-			if ((uint32_t)flow_tag_id_candidate != flow_tag_id) {
-				// tag_id is out of the range by mask, will not use it
-				ring_logdbg("flow_tag disabled as tag_id: %d is out of mask (%x) range!",
-					    flow_tag_id, FLOW_TAG_MASK);
-				flow_tag_id = FLOW_TAG_MASK;
-			}
-			ring_logdbg("sock_fd:%d enabled:%d with id:%d",
-				    flow_tag_id_candidate-1, m_flow_tag_enabled, flow_tag_id);
-		} else {
+		if (!flow_tag_id) {
 			flow_tag_id = FLOW_TAG_MASK; // FLOW_TAG_MASK - modal, FT to be attached but will not be used
-			ring_logdbg("flow_tag:%d disabled as flow_tag_id_candidate:%d", flow_tag_id, flow_tag_id_candidate);
+			ring_logdbg("flow_tag:%d disabled", flow_tag_id);
```

Comment:
I think it is useless code as far as `flow_tag_id` is defined on sockinfo layer and here it should be guaranteed as >0 or FLOW_TAG_MASK.